pkg/server/handler.go

```diff
@@ -234,7 +234,100 @@ func WithInClusterServiceAccountRequestRewrite(handler http.Handler, unsafeServi
 	})
 }
 
-func mergeCRDsIntoCoreGroup(crdLister v1.CustomResourceDefinitionLister, crdHandler, coreHandler func(res http.ResponseWriter, req *http.Request)) restful.FilterFunction {
+func WithWildcardIdentityGuard(handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		cluster := request.ClusterFrom(req.Context())
+		if cluster != nil && !cluster.Wildcard {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		requestInfo, ok := request.RequestInfoFrom(req.Context())
+		if !ok {
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("missing requestInfo")),
+				errorCodecs, schema.GroupVersion{}, w, req,
+			)
+			return
+		}
+		if !requestInfo.IsResourceRequest {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		// SUPER HACK
```

Comment:
don't we have a list of system CRDs elsewhere? I think I also saw here in the PR that `kcp.dev` is taken as hint that it is a system CRD.

Reply:
We do have a list, but it's grouped by workspace type. That's why I took this approach.

And I later wrote the code you reference, to just check for kcp.dev. I think I prefer that approach plus admission to reject kcp.dev anywhere other than system:system-crds.

Reply:
Actually, I can remove this check. It's not needed. I originally had it because this handler was going to reject wildcard requests that didn't have an identity, but now that identity is optional, we can remove it.